### PR TITLE
Add Firebase Custom Server Integration.

### PR DESCRIPTION
**Pull request by jefrydco**
Add firebase custom server integration and some conditional to package manager build setup instruction in README.md and sao.js for better description.

### CODE REVIEW

File: template/nuxt/nuxt.config.js [hunk 1]
```diff
@@ -64,7 +62,12 @@ module.exports = {
   */
   axios: {
     // See https://github.com/nuxt-community/axios-module#options
-  },<% } %>
+  },<% } %><% if (server === 'firebase') { %>
+
+  /*
+  ** Build directory
+  */
+  buildDir: '../functions/.nuxt',<% } %>
```
Comment (jefrydco) [inline template/nuxt/nuxt.config.js]: `npm run dev` still doesn't work as expected when any UI framework's custom template is used. Error displayed on the browser console,

> The client-side rendered virtual DOM tree is not matching server-rendered content. This is likely caused by incorrect HTML markup, for example nesting block-level elements inside <p>, or missing <tbody>. Bailing hydration and performing full client-side render.

**Comment by ekoeryanto** [comment]
@davidroyer has succeed with his way, I hope he would like to help here.

**Comment by jefrydco** [comment]
@ekoeryanto I also use code taken from @davidroyer repository. I've tried both my way and david way, it becomes a problem when we use custom tag component from third party library like Vuetify.

File: sao.js [hunk 2]
```diff
@@ -18,6 +20,19 @@ const move = (from, to = '') => {
   return result
 }
 
+const installDependency = (pm, folderPath, folderName = '') => {
```
[the remark this answers is not in the document]
Comment (jefrydco) [reply sao.js]: Hi @clarkdo , you're welcome

Ahh I seee, thanks for the suggestion, I'll try open a pr there as well.

[the remark this answers is not in the document]
Comment (clarkdo) [reply sao.js]: Really sorry for the late reply, actually I reviewed this pr when it was just opened, but I forgot to click the `submit` button 👎 , sorry.

**Comment by PeteyRev** [comment]
@clarkdo @jefrydco Any chance on this getting pushed soon? That would be awesome

**Comment by Atinux** [comment]
Would be great to update the PR to fix these conflicts (or create a new one?)

**Comment by Atinux** [comment]
I do believe it's best to use [nuxt-fire](https://github.com/nuxt-community/firebase-module) for such integration with Firebase.

**Pull request closed without merging**